CRM/Utils/System/Drupal8.php

```diff
@@ -419,8 +419,8 @@ public function loadBootStrap($params = [], $loadUser = TRUE, $throwError = TRUE
     $kernel->preHandle($request);
     $container = $kernel->rebuildContainer();
     // Add our request to the stack and route context.
-    $request->attributes->set(\Symfony\Cmf\Component\Routing\RouteObjectInterface::ROUTE_OBJECT, new \Symfony\Component\Routing\Route('<none>'));
-    $request->attributes->set(\Symfony\Cmf\Component\Routing\RouteObjectInterface::ROUTE_NAME, '<none>');
+    $request->attributes->set(\Drupal\Core\Routing\RouteObjectInterface::ROUTE_OBJECT, new \Symfony\Component\Routing\Route('<none>'));
+    $request->attributes->set(\Drupal\Core\Routing\RouteObjectInterface::ROUTE_NAME, '<none>');
```

Comment:
Part of me says, "Yeah, D8 is EOL, and the only people who've mentioned it (to me) have said they use D9."

OTOH, I haven't *really* asked around, and we kept D6 running for years after EOL, and I do think there's a sort of procedural obligation -  e.g. we should clearly announce in advance. I'd maybe more inclined to say:

* In 5.50 and 5.51 (*the next ESR*), show a big warning that D8 will break soon. (*This would give de-facto 6-7 month offramp.*)
* In 5.52, you can do the actual break.

From a near-term perspective, on this specific issue, it may be more expedient to say:

```suggestion
    $routeInterface = class_exists('\Drupal\Core\Routing\RouteObjectInterface')
      ? '\Drupal\Core\Routing\RouteObjectInterface' 
      : '\Symfony\Cmf\Component\Routing\RouteObjectInterface';
    $request->attributes->set($routeInterface::ROUTE_OBJECT, new \Symfony\Component\Routing\Route('<none>'));
    $request->attributes->set($routeInterface::ROUTE_NAME, '<none>');
```

Reply:
It's a small thing to do in this PR but I'm going to try to make a stronger argument for not even needing to think about it:

* There's the 3 reasons already mentioned:
  * Has been EOL for 6 months
  * No security updates.
    * And there have been recent drupal security fixes which would have helped prompt people to move off 8 if they were still on it.
  * Civi works equally as well on 9 as on 8
* There's support for dropping it in chat, with nobody dissenting: https://chat.civicrm.org/civicrm/pl/qh6t761dt7gsdyp78m6stbmpwo
* Drupal 8 to 9 is _very_ different than 6 to 7. Drupal 9 is identical to drupal 8.9 except for some deprecated functions. It's not much different than if they had called it 8.10.
  * There were many blog posts about this on the internet, about how easy it would be to upgrade compared to other jumps.
* There was plenty of awareness ahead of time in the drupal world that drupal 8 would be unsupported.
  * From May 2021 to November 2021, the only updates for drupal 8.9.x were security updates, with repeated notices that 8 was becoming EOL.
  * There was a module available that was pretty good about pointing out what if anything you needed to change.
* There is no CIVICRM_UF=Drupal9, because it's not needed.